roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3571 @@
+    return param->getVal();
+  }
+  return 0;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// set the morphing parameters to those supplied in the given param hist
+
+void RooLagrangianMorphFunc::setParameters(TH1 *paramhist) {
+  setParams(paramhist, this->_operators, false);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// set the morphing parameters to those supplied in the sample with the given
+/// name
+
+void RooLagrangianMorphFunc::setParameters(const char *foldername) {
+  std::string filename = this->_config.getFileName();
+  TDirectory *file = openFile(filename.c_str());
+  TH1 *paramhist = getParamHist(file, foldername);
+  setParams(paramhist, this->_operators, false);
+  closeFile(file);
+}
+
+/////////////////////////////////////////////////////////////////////////////////
+/// retrieve the morphing parameters associated to the sample with the given
+/// name
+
+RooLagrangianMorphFunc::ParamSet
+RooLagrangianMorphFunc::getParameters(const char *foldername) const {
+  const std::string name(foldername);
+  return this->_config.getParamCards().at(name);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// set the morphing parameters to those supplied in the list with the given
+/// name
+
+void RooLagrangianMorphFunc::setParameters(const RooArgList *list) {
+  for (auto itr : *list) {
+    RooRealVar *param = dynamic_cast<RooRealVar *>(itr);
+    if (!param)
+      continue;
+    this->setParameter(param->GetName(), param->getVal());
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// retrieve the histogram observable
+
+RooRealVar *RooLagrangianMorphFunc::getObservable() const {
+  if (this->_observables.getSize() < 1) {
+    coutE(InputArguments) << "observable not available!" << std::endl;
+    return NULL;
+  }
+  return static_cast<RooRealVar *>(this->_observables.at(0));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// retrieve the histogram observable
+
+RooRealVar *RooLagrangianMorphFunc::getBinWidth() const {
+  if (this->_binWidths.getSize() < 1) {
+    coutE(InputArguments) << "bin width not available!" << std::endl;
+    return NULL;
+  }
+  return static_cast<RooRealVar *>(this->_binWidths.at(0));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// retrieve a histogram output of the current morphing settings
+
+TH1 *RooLagrangianMorphFunc::createTH1(const std::string &name,
+                                       RooFitResult *r) {
+  return this->createTH1(name, false, r);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// retrieve a histogram output of the current morphing settings
+
+TH1 *RooLagrangianMorphFunc::createTH1(const std::string &name,
+                                       bool correlateErrors, RooFitResult *r) {
```

Comment:
Why does this `createTH1` function take a `RooFitResult *` just to `delete` it but doesn't do anything with it? You can keep the `RooFitResult *r` parameter if you intend to use it when you update the code that's fine, but don't assume ownership (it's also better to take it by reference if it can't be `nullptr`).